### PR TITLE
New frontend

### PR DESCRIPTION
**Pull request by Wzixiao**
## description

In this PR, we mainly did three things:

1. Created a new front-end project and placed it in the root project (Only the simplest request was sent and displayed)

2. Changed the project name for the front-end and back-end.

3. Optimized the steps for starting the project (Docker startup not yet provided)

## Code change

```
.
├── README.md # update start flow
├── bin
│   └── migrate.bat # new added windwos shell script
├── package-lock.json  # This file is reserved for the purpose of unifying the dependency format of various developers on the front-end
├── package.json  # node.js requirements file
├── requirements.txt # new added 'celery'、'flask-cors' requirements
├── smart_html_client # change dir name(frontend)
└── smart_html_server # change dir name(backend)
  └── engine
     └── openai.py # new added '_find_between' static function
```

Due to modifying the```html``` function of ```OpenAIResult```, the previous backend page cannot be used.

I don't know what version ```celery``` should use, so I typed TODO, which is a dependency left behind in the master

### CODE REVIEW

File: smart_html_server/main.py
Comment (Wzixiao) [inline smart_html_server/main.py]: This is because the main we started is not in the root directory, which can cause ```__Init__``` Invalid, so I think we can do this for now?

Comment (dyzsasd) [reply smart_html_server/main.py]: I find we are using npm to run the dev server locally, is it possible to set PYTHONPATH environment variable in node script instead of setting here. We may need to use WSGI server in prod instead of flask buildin server

File: bin/migrate.sh
Comment (dyzsasd) [inline bin/migrate.sh:8]: It's better to keep the newline here

File: requirements.txt
Comment (dyzsasd) [inline requirements.txt]: My bad, a fix has been added in master

File: requirements.txt
Comment (dyzsasd) [inline requirements.txt]: Why do we need cross domain access

File: smart_html_server/core/engine/openai.py
Comment (dyzsasd) [inline smart_html_server/core/engine/openai.py:48]: Good

File: smart_html_client/src/components/PromptInput/PromptInput.tsx
Comment (dyzsasd) [inline smart_html_client/src/components/PromptInput/PromptInput.tsx:11]: Can the text input area be adjusted in size automatically based on the input content?

**Comment by dyzsasd** [comment]
In this project, it seems we have two separate services. For now, we can temporarily manage both services using npm, but it's advisable to note this as a TODO item. My suggestion is to eventually set up both services locally using docker-compose and nginx, mirroring our production environment setup.

**Comment by dyzsasd** [comment]
What should be changed in docker file for adding the npm code, maybe we need to just create a new docker image.

File: smart_html_client/src/components/PromptInput/PromptInput.tsx
Comment (chenJuZ) [reply smart_html_client/src/components/PromptInput/PromptInput.tsx:11]: image
We are using a textarea input field, which supports multi-line input and allows line breaks with Shift+Enter.

File: requirements.txt
Comment (chenJuZ) [reply requirements.txt]: Your understanding is correct (although, in my view, CORS is usually configured on the backend).

When we develop React applications, we typically use a front-end server (like Vite or Webpack), commonly referred to as a "development server" or "hot reload server." 

This development server can monitor changes in the project files and automatically reload the page or replace the modified modules without manually refreshing the browser. 

When multiple servers are deployed on different ports and communicate through APur understanding is correct (although, in my view, CORS is usually configured on the backend).

When we develop React applications, we typically use a front-end server (like Vite or Webpack), commonly referred to as a "development server" or "hot reload server." 

Currently, we handle CORS on the backend, but considering our plans for production deployment (where Nginx will resolve CORS issues in the production stage), we will later configure CORS on the front-end instead of the backend.

File: smart_html_server/main.py
Comment (Wzixiao) [reply smart_html_server/main.py]: You are right, using environment variables is a better choice.

But even if ASGI/WSGI is used, I think this problem will still not be solved during development (perhaps just moving the problem to the ASGI/WSGI server, of course, if ASGI/WSGI server script is in the root directory, this problem will not occur)

File: smart_html_client/src/components/PromptInput/PromptInput.tsx
Comment (Wzixiao) [reply smart_html_client/src/components/PromptInput/PromptInput.tsx:11]: I used chatgpt as a reference, although they may be slightly different.

**Comment by chenJuZ** [comment]
> What should be changed in docker file for adding the npm code, maybe we need to just create a new docker image.



> What should be changed in docker file for adding the npm code, maybe we need to just create a new docker image.

I feel that using Docker for front-end deployment in a production environment is not very practical, especially when it comes to the need for real-time deployment and updating of pages, which can be challenging. I've attempted to use volumes for the build process, but realistically, this approach is not quite standard. We might provide a production environment setup using docker-compose, as you suggested, where we would use Nginx for serving the front-end.

**Comment by chenJuZ** [comment]
> What should be changed in docker file for adding the npm code, maybe we need to just create a new docker image.

I feel that using Docker for front-end deployment in a production environment is not very practical, especially when it comes to the need for real-time deployment and updating of pages, which can be challenging. I've attempted to use volumes for the build process, but realistically, this approach is not quite standard. We might provide a production environment setup using docker-compose, as you suggested, where we would use Nginx for serving the front-end.

File: smart_html_server/main.py
Comment (dyzsasd) [reply smart_html_server/main.py]: Understood. It seems necessary to modify the main function, as its current form is not suitable for a Flask application in a production environment. I'll substitute it with a dev_server.py instead.

File: requirements.txt
Comment (dyzsasd) [reply requirements.txt]: Indeed, the primary concern is the approach to operating a development server versus a production server; they ought to be distinct. I'll implement the required modifications following the merger of this pull request.

File: smart_html_client/src/components/PromptInput/PromptInput.tsx
Comment (dyzsasd) [reply smart_html_client/src/components/PromptInput/PromptInput.tsx:11]: Oh, find it, nice

**Comment by dyzsasd** [comment]
The unit test failed due to the GCP issue, you can merge this PR, I will make a fix after this PR.

**Comment by Wzixiao** [comment]
Hi @dyzsasd 
I think the current version can be merged, what do you think?

**Comment by Wzixiao** [comment]
> What should be changed in docker file for adding the npm code, maybe we need to just create a new docker image.

yeah, it is easiest way

File: smart_html_server/main.py
Comment (Wzixiao) [reply smart_html_server/main.py]: We can complete ```dev_server.py ``` later,, but the current environment variables means are definitely much better than my previous method.

File: requirements.txt
Comment (Wzixiao) [reply requirements.txt]: Our current strategy:

No backend code used at all 
- development
    We use a front-end proxy
- production
    We use nginx

**Comment by dyzsasd** [comment]
> Hi @dyzsasd I think the current version can be merged, what do you think?

It's approved, you can merge it when you think it's ok, I will rebase my branch and modify the CI.

**Review by dyzsasd — APPROVED** [review]

**Comment by Wzixiao** [comment]
> > Hi @dyzsasd I think the current version can be merged, what do you think?
> 
> It's approved, you can merge it when you think it's ok, I will rebase my branch and modify the CI.

Oh no, I'm not depository member, so I don't have a merge button for me to click.

**Comment by dyzsasd** [comment]
> > > Hi @dyzsasd I think the current version can be merged, what do you think?
> > 
> > 
> > It's approved, you can merge it when you think it's ok, I will rebase my branch and modify the CI.
> 
> Oh no, I'm not depository member, so I don't have a merge button for me to click.

Just added you in the project, you can add other people

**Pull request merged**
